### ISSUE TITLE
Crash: decayed attributed array

### ISSUE THREAD

**Issue by ehaas** [Vexu/arocc#237]
I believe we might need a `decayed_attributed` type specifier - currently, decaying an attributed array turns it into a `.special_va_start`

```c
__attribute__((aligned)) int foo[1] = {0};
int *bar = foo;
```

```
thread 9698834 panic: reached unreachable code
/Users/ehaas/source/arocc/src/Type.zig:1427:21: 0x10399d5df in Type.Builder.fromType (arocc)
            else => unreachable,
                    ^
/Users/ehaas/source/arocc/src/Type.zig:1515:48: 0x10396060c in Type.printPrologue (arocc)
        else => try w.writeAll(Builder.fromType(ty).str().?),
                                               ^
/Users/ehaas/source/arocc/src/Type.zig:1448:29: 0x10395f8fa in Type.print (arocc)
    _ = try ty.printPrologue(w);
```

**Comment by Vexu** [comment]
This no longer crashes but the result is incorrect:
```yaml
var: 'attributed([1]int)'
 attr: aligned alignment: null
 name: foo
 init:
  array_init_expr_two: '[1]int'
    int_literal: 'int' (value: 0)

var: '*int'
 name: bar
 init:
  implicit_cast: (null_to_pointer) '*int'
    implicit_cast: (array_to_pointer) 'nullptr_t' # why?
      decl_ref_expr: 'attributed([1]int)' lvalue
       attr: aligned alignment: null
       name: foo

```

**Comment by ehaas** [comment]
Closed by #568
